src/popup/components/OutOfFunds.tsx

```diff
@@ -1,76 +1,198 @@
 import React from 'react'
-import { Button } from './ui/Button'
-import type { RecurringGrant, OneTimeGrant } from '@/shared/types'
+import { useForm } from 'react-hook-form'
+import type { RecurringGrant, OneTimeGrant, AmountValue } from '@/shared/types'
 import type { AddFundsPayload, Response } from '@/shared/messages'
 import type { WalletAddress } from '@interledger/open-payments'
-import { getCurrencySymbol, transformBalance } from '../lib/utils'
+import {
+  charIsNumber,
+  formatNumber,
+  getCurrencySymbol,
+  transformBalance
+} from '@/popup/lib/utils'
+import { useTranslation } from '@/popup/lib/context'
 import { getNextOccurrence } from '@/shared/helpers'
+import { ErrorMessage } from '@/popup/components/ErrorMessage'
+import { Button } from '@/popup/components/ui/Button'
+import { Input } from '@/popup/components/ui/Input'
 
 interface OutOfFundsProps {
   info: Pick<WalletAddress, 'id' | 'assetCode' | 'assetScale'>
   grantRecurring?: RecurringGrant['amount']
   grantOneTime?: OneTimeGrant['amount']
-  requestAddFunds: (details: AddFundsPayload) => Promise<Response>
+  onChooseOption: (recurring: boolean) => void
 }
 
 export const OutOfFunds = ({
   info,
   grantOneTime,
   grantRecurring,
-  requestAddFunds
+  onChooseOption
 }: OutOfFundsProps) => {
   if (!grantOneTime && !grantRecurring) {
     throw new Error('Provide at least one of grantOneTime and grantRecurring')
   }
+  const t = useTranslation()
 
-  const currencySymbol = getCurrencySymbol(info.assetCode)
-  const amount = transformBalance(
-    grantRecurring?.value || grantOneTime!.value,
-    info.assetScale
+  return (
+    <div className="flex flex-col gap-4">
+      <ErrorMessage
+        error={t('outOfFunds_error_title')}
+        className="mb-0 text-error"
+      />
+      <div className="px-2 text-xs text-medium">
+        <p>{t('outOfFunds_error_text')}</p>
+        <p>{t('outOfFunds_error_textHint')}</p>
+        {grantRecurring?.value && (
+          <p className="mt-1">
+            <RecurringAutoRenewInfo
+              info={info}
+              grantRecurring={grantRecurring}
+            />
+          </p>
+        )}
+      </div>
+
+      <div className="w-100 h-0.5 bg-disabled" />
+
+      <Button onClick={() => onChooseOption(true)}>
+        {t('outOfFunds_action_optionRecurring')}
+      </Button>
+      <Button onClick={() => onChooseOption(false)}>
+        {t('outOfFunds_action_optionOneTime')}
+      </Button>
+    </div>
   )
+}
 
-  const requestTopUpOneTime = () => {
-    return requestAddFunds({ amount: amount, recurring: false })
-  }
+interface AddFundsProps {
+  info: Pick<WalletAddress, 'id' | 'assetCode' | 'assetScale'>
+  recurring: false | 'P1M'
```

Comment:
Why is this `false | P1M`?

Reply:
In future we can support different durations. False means one time, rest are periods in ISO notation. 
Should probably just stick to Boolean for now 😄